### ISSUE TITLE
renderToReadableStream function throws an error in Next.js 14

### ISSUE THREAD

**Issue by Mriganka5137** [resend/react-email#1029]
### renderToReadableStream function throws an error in Next.js 14
When using react-email in next.js 14 app Found the following bug:
`./node_modules/@react-email/render/dist/index.mjs
Attempted import error: 'renderToReadableStream' is not exported from 'react-dom/server' (imported as 'renderToReadableStream').`

I have to modify the following code in the @react-email package 
```tsx
import { renderToPipeableStream, renderToStaticMarkup as renderToStaticMarkup2 } from "react-dom/server";

async function renderToString(children) {
  const stream = await renderToPipeableStream(children)`
```
![bug-react-mail](https://github.com/resendlabs/react-email/assets/105497106/981fa96e-3f72-4153-9799-714c1d6df41e)
![bug-react-mail2](https://github.com/resendlabs/react-email/assets/105497106/0215de1b-fba7-4a00-a8a7-c67460ab2ba8)

 The renderToReadableStream function is not compatible with Node.js, as it requires Web Streams support. For Node.js, you should use the renderToPipeableStream function instead.

### Which package is affected (leave empty if unsure)

@react-email/render

### Link to the code that reproduces this issue

https://github.com/Mriganka5137/react-email-bug

### To Reproduce

I just followed all the steps that were mentioned in the react-email docs.

### Expected Behavior

The expected behaviour is the react-email client on localhost:3030 port

### What's your node version? (if relevant)

v21.1.0

**Comment by gabrielmfern** [comment]
This issues should've been fixed with `@react-email/render@0.0.9` which is on `@react-email/components@0.0.11`.
Is this problem happening on build? Also, I tried cloning your repo and build seems to work just fine. Maybe clearing the dependencies and the lock and reinstalling might fix the issue here?

**Comment by inventive-username** [comment]
I'm having no luck with `@react-email/render@0.0.9` - @gabrielmfern do you have any suggestions on how to resolve this?

**Comment by Mriganka5137** [comment]
> I'm having no luck with `@react-email/render@0.0.9` - @gabrielmfern do you have any suggestions on how to resolve this?

I had to manually replace 'renderToReadableStream' with 'renderToPipeableStream' in the node_modules of  ' .react-email >node_modules > @react-email'

**Comment by gabrielmfern** [comment]
@Mriganka5137, @inventive-username can you guys try deleting the `.react-email` folder and trying again?

**Comment by inventive-username** [comment]
@gabrielmfern unfortunately my CI/CD pipeline setup does not allow me to manually update node_modules - however, I found that downgrading from next 14 to next 13 worked for me 👍

**Comment by gabrielmfern** [comment]
> @gabrielmfern unfortunately my CI/CD pipeline setup does not allow me to manually update node_modules - however, I found that downgrading from next 14 to next 13 worked for me 👍

Not sure if this is true, but maybe you aren't git ignoring the `.react-email` folder? It should be ignored generally

**Comment by aadamw** [comment]
Hey @Mriganka5137 
I've found a workaround for this issue. You need to install the @react-email/render: 0.0.9 manually

```bash
pnpm install   @react-email/render
```

**Comment by petergombos** [comment]
Works fine in API routes but I get the following when using it in a server function:

<img width="1016" alt="Screenshot 2023-11-19 at 21 45 44" src="https://github.com/resendlabs/react-email/assets/1838896/e5b53c13-5415-4ad7-bcdb-d327ecd2b5fb">
<img width="534" alt="Screenshot 2023-11-19 at 21 54 59" src="https://github.com/resendlabs/react-email/assets/1838896/88e10ca5-aaf5-4c1c-9806-aea4de8f753a">

**Comment by ryanhefner** [comment]
> Hey @Mriganka5137 I've found a workaround for this issue. You need to install the @react-email/render: 0.0.9 manually
> 
> ```shell
> pnpm install   @react-email/render
> ```

This fixed it for me.

**Comment by Mriganka5137** [comment]
> > Hey @Mriganka5137 I've found a workaround for this issue. You need to install the @react-email/render: 0.0.9 manually
> > 
> > ```shell
> > pnpm install   @react-email/render
> > ```
> 
> This fixed it for me.

Thank you so much. Worked for me too.

**Comment by cipriancaba** [comment]
`@react-email/render` gets resolved to various different versions. Please pin it to the latest in all the components that use it in the next release

![Screenshot 2023-11-22 at 18 30 48](https://github.com/resendlabs/react-email/assets/695515/b08f6813-c83e-48f7-9b31-37a4155a875a)

**Comment by marketingead** [comment]
I was experiencing the same problem `Attempted import error: 'renderToReadableStream' is not exported from 'react-dom/server' (imported as 'renderToReadableStream')` using `"next": "^14.0.3",`.
I managed to solve it by removing the `@react-email` folder from the node_modules folder and reinstalling the packages manually in their latest versions (`"@react-email/render": "^0.0.9"`, `"@react-email/ preview": "0.0.7",` `"@react-email/components": "^0.0.11",`).

#1046 may help fix the issue.

**Comment by eduardobernardo** [comment]
I was experiencing the same problem `Attempted import error: 'renderToReadableStream' is not exported from 'react-dom/server' (imported as 'renderToReadableStream')` using `"next": "^14.0.3",`.

I managed to solve it by removing the `@react-email` folder from the node_modules folder and reinstalling the packages manually in their latest versions (`"@react-email/render": "^0.0.9"`, `"@react-email/ preview": "0.0.7",` `"@react-email/components": "^0.0.11",`).

#1046 may help fix the issue.

**Comment by jamesvclements** [comment]
Manually installing @react-email/render per @aadamw's fixed it for me as well

```bash
npm i @react-email/render
```

I'm on:
```json
"dependencies": {
  "@react-email/components": "0.0.12",
  "@react-email/render": "^0.0.10",
  "react-email": "1.10.0",
  "resend": "2.0.0",
}
```

**Comment by RoyBkker** [comment]
Manually installing @react-email/render per @aadamw's fixed it for me as well. Thanks mate

**Comment by gabrielmfern** [comment]
This has been fixed ever since `@react-email/render@1.0.0`
